frontend/src/pages/ProductsPage/ProductsPage.js

```diff
@@ -2,45 +2,51 @@ import Table from "../../components/Admin/Table/Table"
 import { useProducts } from "../../hooks/product-hooks"
 import { useMemo } from "react"
 import { Link } from "react-router-dom"
+import { humanReadableDate } from "../../helpers/helper"
+import TableSkeleton from "../../components/Admin/TableSkeleton/TableSkeleton"
 
 const ProductsPage = () => {
-    const products = useProducts()
-    const columns = useMemo(
-        () => [
-            {
-                Header: "Products",
-                columns: [
-                    {
-                        Header: "ID",
-                        accessor: "id",
-                    },
-                    {
-                        Header: "Name",
-                        accessor: "name",
-                    },
-                    {
-                        Header: "Price",
-                        accessor: "price",
-                    },
-                    {
-                        Header: "Created At",
-                        accessor: "createdAt",
-                    },
-                ],
-            },
+  const products = useProducts()
+  const columns = useMemo(
+    () => [
+      {
+        Header: "Products",
+        columns: [
+          {
+            Header: "ID",
+            accessor: "id",
+          },
+          {
+            Header: "Name",
+            accessor: "name",
+          },
+          {
+            Header: "Price",
+            accessor: "price",
+          },
+          {
+            Header: "Created At",
+            accessor: "createdAt",
+            Cell: ({ cell: { value } }) => humanReadableDate(value),
+          },
         ],
-        []
-    )
+      },
+    ],
+    []
+  )
 
-    if (products.isError) return <h1>Something went wrong!</h1>
-    if (products.isLoading) return <h1>Loading...</h1>
+  if (products.isError) return <h1>Something went wrong!</h1>
 
-    return (
-        <div className="bg-white h-full overflow-y-scroll">
-            <Link className="admin-form-submit w-24 ml-2" to="/admin/products/create">+ Add</Link>
-            <Table columns={columns} data={products?.data?.data?.data?.products} />
-        </div>
-    )
+  return (
+    <div className="bg-white h-full overflow-y-scroll">
+      <Link className="admin-form-submit w-24 ml-2" to="/admin/products/create">+ Add</Link>
+      {products.isLoading ? (
+        <TableSkeleton />
+      ) : (
+        <Table columns={columns} data={products?.data?.data?.data?.products} />
+      )}
```

Comment:
I think the parentheses aren't needed and probably should adopt the previous style ```if (products.isLoading) return <h1>Loading...</h1>```

Reply:
Unfortunately adopting the previous method would make the pages look funny. The way that we're handling the tailwind styles right now is within the large return block at the end with `<div className="bg-white h-full overflow-y-scroll">`. If either the `TableSkeleton` `isLoading` or `isError` blocks are outside of that, things would not follow the layout that we currently have set. This also means that since the <Link> is within that final block as well, loading + error state would prevent the add button from ever loading. 
I've made a change that should satisfy the cases where the data is loading/erroring and still allow the page to render properly.

I moved ALL table content outside of that section and placed it in it's own function like this:

```
  const content = useMemo(
    () => {
      if (products.isError) return <h1>Something went wrong!</h1>
      if (products.isLoading) return <TableSkeleton />
      return <Table columns={columns} data={products?.data?.data?.data?.products} />
    }, [products]
  )
```

This should mean the actual render will be much cleaner
```
  return (
    <div className="bg-white h-full overflow-y-scroll">
      <Link className="admin-form-submit w-24 ml-2" to="/admin/products/create">+ Add</Link>
      {content}
    </div>
  )
```